pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -178,6 +195,17 @@ type Client interface {
 	// ExecutePreProvisionCommand executes a command before the provision process starts.
 	// A non-zero exit status will indicate that provisioning should not start.
 	ExecutePreProvisionCommand(ctx context.Context, preProvisionCommand string) (exitStatus int, stdoutAndStderr string, err error)
+
+	// StartImageURLCommand calls the command provided via image-url-command.
+	// It gets called by the controller after the rescue system of the new machine
+	// is reachable. The env var `OCI_REGISTRY_AUTH_TOKEN` gets set to the same value of the
+	// corresponding env var of the controller.
+	// This gets used when imageURL set.
+	// For hcloud deviceNames is always {"sda"}. For baremetal it corresponds to the WWNs
+	// of RootDeviceHints.
+	StartImageURLCommand(ctx context.Context, command, imageURL string, bootstrapData []byte, machineName string, deviceNames []string) (exitStatus int, stdoutAndStderr string, err error)
+
+	StateOfImageURLCommand() (state ImageURLCommandState, logFile string, err error)
```

Comment:
can you add a description of this? I don't understand this function

Reply:
I added a comment:

```go 
	// StateOfImageURLCommand returns the current states of the ImageURLCommand. States can
	// be: NotStarted, Running, Failed, FinishedSuccesfully.
```

ok?

Reply:
yes